src/transport/PeerMessageCounter.h

```diff
@@ -0,0 +1,159 @@
+/*
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file defines the CHIP message counters of remote nodes.
+ *
+ */
+#pragma once
+
+#include <array>
+#include <bitset>
+
+namespace chip {
+namespace Transport {
+
+class PeerMessageCounter
+{
+public:
+    static constexpr size_t kChallengeSize = 8;
+
+    PeerMessageCounter() : mStatus(Status::NotSync) {}
+    ~PeerMessageCounter() {}
+
+    void Reset() { mStatus = Status::NotSync; }
+
+    bool IsSyncStarted() { return mStatus != Status::NotSync; }
+    bool IsSynchronizing() { return mStatus == Status::SyncInProcess; }
+    bool IsSyncCompleted() { return mStatus == Status::Synced; }
+
+    void StartSync(std::array<uint8_t, kChallengeSize> challenge)
```

Comment:
```suggestion
    void SyncStarting(std::array<uint8_t, kChallengeSize> challenge)
```
This is a notification, not a command.....